Option allowed-redefined-builtins has no effect

### Bug description

The option allowed-redefined-builtins appears to have no effect.

Here is an example file `a.py`:

```python
id = 1
```

### Command used

```shell
pylint --allowed-redefined-builtins id a.py
```

### Pylint output

```
************* Module a
a.py:2:0: C0305: Trailing newlines (trailing-newlines)
a.py:1:0: C0114: Missing module docstring (missing-module-docstring)
a.py:1:0: W0622: Redefining built-in 'id' (redefined-builtin)
a.py:1:0: C0103: Constant name "id" doesn't conform to UPPER_CASE naming style (invalid-name)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

The message `a.py:1:0: W0622: Redefining built-in 'id' (redefined-builtin)` should not appear.

### Pylint version

```shell
pylint 3.3.3
astroid 3.3.8
Python 3.13.1 (main, Dec  4 2024, 18:05:56) [GCC 14.2.1 20240910]
```

--- a/doc/data/messages/r/redefined-builtin/details.rst
+++ b/doc/data/messages/r/redefined-builtin/details.rst
@@ -1,0 +1,23 @@
+The `allowed-redefined-builtins <https://pylint.readthedocs.io/en/stable/user_guide/configuration/all-options.html#allowed-redefined-builtins>`_
+option lets you specify names that are permitted to shadow built-ins.
+
+However, this option is not effective for redefinitions at the module level or for global variables. For example:
+
+Module-Level Redefinitions::
+
+    # module_level_redefine.py
+    id = 1  # Shadows the built-in `id`
+
+Global Variable Redefinitions::
+
+    # global_variable_redefine.py
+    def my_func():
+        global len
+        len = 1  # Shadows the built-in `len`
+
+Rationale:
+
+Shadowing built-ins at the global scope is discouraged because it obscures their behavior
+throughout the entire module, increasing the risk of subtle bugs when the built-in is needed elsewhere.
+In contrast, local redefinitions are acceptable as their impact is confined to a specific scope,
+reducing unintended side effects and simplifying debugging.

--- a/doc/data/messages/r/redefined-builtin/details.rst
+++ b/doc/data/messages/r/redefined-builtin/details.rst
@@ -1,5 +1,4 @@
-The `allowed-redefined-builtins <https://pylint.readthedocs.io/en/stable/user_guide/configuration/all-options.html#allowed-redefined-builtins>`_
-option lets you specify names that are permitted to shadow built-ins.
+The :ref:`allowed-redefined-builtins <variables-options>` option lets you specify names that are permitted to shadow built-ins.
 
 However, this option is not effective for redefinitions at the module level or for global variables. For example:
 

--- a/tests/functional/r/redefined/redefined_builtin_allowed.py
+++ b/tests/functional/r/redefined/redefined_builtin_allowed.py
@@ -7,3 +7,8 @@ def function():
     print(dir, dict)
 
 list = "not in globals" # [redefined-builtin]
+
+def global_variable_redefine():
+    """Shadow `len` using the `global` keyword."""
+    global len
+    len = 1  # [redefined-builtin]

--- a/tests/functional/r/redefined/redefined_builtin_allowed.rc
+++ b/tests/functional/r/redefined/redefined_builtin_allowed.rc
@@ -1,4 +1,4 @@
 [messages control]
-disable = invalid-name
+disable = invalid-name, global-variable-undefined
 [variables]
-allowed-redefined-builtins = dir, list
+allowed-redefined-builtins = dir, list, len

--- a/tests/functional/r/redefined/redefined_builtin_allowed.txt
+++ b/tests/functional/r/redefined/redefined_builtin_allowed.txt
@@ -1,2 +1,3 @@
 redefined-builtin:6:4:6:8:function:Redefining built-in 'dict':UNDEFINED
 redefined-builtin:9:0:9:4::Redefining built-in 'list':UNDEFINED
+redefined-builtin:14:4:14:7:global_variable_redefine:Redefining built-in 'len':UNDEFINED

